mon-pix/app/controllers/assessments/challenge.js

```diff
@@ -69,16 +69,14 @@ export default class ChallengeController extends Controller {
   }
 
   @action
-  async setFocusedOutOfChallenge(value) {
+  setFocusedOutOfChallenge(value) {
     this.hasFocusedOutOfChallenge = value;
-    if (this.hasFocusedOutOfChallenge) {
-      await this.model.assessment.save({ adapterOptions: { updateLastQuestionsState: true, state: 'focusedout' } });
-    }
   }
 
   @action
-  focusedOutOfWindow() {
+  async focusedOutOfWindow() {
     this.hasFocusedOutOfWindow = true;
+    await this.model.assessment.save({ adapterOptions: { updateLastQuestionsState: true, state: 'focusedout' } });
```

Comment:
```suggestion
    if (this.hasFocusedOutOfChallenge) {
      await this.model.assessment.save({ adapterOptions: { updateLastQuestionsState: true, state: 'focusedout' } });
    }
```
On n'a pas besoin de sauvegarder le focusedout pour les non focus

Reply:
`if (this.model.challenge.focused) {` plutot non ?

Reply:
oui pardon ^^

Reply:
Il se trouve que de toute façons ce callback n'est jamais appelé si ce n'est pas une question focus.